src/Microsoft.Azure.WebJobs.Host/Exceptions/JobMethodException.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Exception that is tied to a specific job method
+    /// </summary>
+    [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")]
+    [Serializable]
+    public class JobMethodException : RecoverableException
```

Comment:
Can you rename this FunctionException? That naming makes sense with the derived FunctionIndexingException and FunctionInvocationException. The latter should also probably derive from your new base class?